Calling nested embedded method `signal 11: segmentation fault`

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version: 0.2.4 b34860e** 
**OS: ArchLinux** 

**What did you do?**
```v
fn main(){
	mut win := &Window{}
	mut ll := &LinearLayout{}
	mut lbl := &Label{
		x: 10
		y: 20
	}
	
	ll.add(mut lbl)
	win.add(mut ll)
	
	win.init()
}

//--------------------

[heap]
pub struct Window{
mut:
	initables []&Initable
}

interface Initable{
mut:
	init(&Window)
}

pub fn (mut w Window) add(mut initable &Initable) {
	// initable.init(&w)
	w.initables << initable
}

interface Container{
mut:
	layout()
}

fn (mut w Window) init(){
	for wd in w.initables{
		if mut wd is Container{
			mut c := wd as Container
			c.layout()
		}
	}
}

//--------------------

pub struct Rect{
mut:
	x      int
	y      int
}

pub fn (r Rect) get_pos() (int, int) {
	return r.x, r.y
}

//--------------------

pub struct LayoutBase{
	Rect
}

//--------------------

pub struct Base{
	LayoutBase
}

pub fn (mut b Base) init(window &Window){}

//--------------------

[heap]
pub struct Label{
	Base
}

//--------------------

pub interface Layoutable{
	get_pos() (int, int)
}

[heap]
pub struct LinearLayout{
	Base
mut:
	layoutables []Layoutable // (2) BUG []&Layoutable -> error: struct or union expected
}

pub fn (mut ll LinearLayout) add(mut l &Layoutable){
	// l.init(ll.get_window())
	ll.layoutables << l
}

pub fn (mut ll LinearLayout) layout() {
	// dump(ll.layoutables.len)
	for mut wl in ll.layoutables{
		// dump(wl)
		x, y := wl.get_pos() // (1) signal 11: segmentation fault
	}
}
```
comment out 1), then works

**What did you expect to see?**
No error.
 
**What did you see instead?**
run this program then (1)
```
signal 11: segmentation fault
/tmp/v_1000/gui.12368496478808544922.tmp.c:6636: at print_backtrace: Backtrace
/tmp/v_1000/gui.12368496478808544922.tmp.c:6702: by v_segmentation_fault_handler
7fb49eacdda0 : by ???
/tmp/v_1000/gui.12368496478808544922.tmp.c:11371: by main__Window_init
/tmp/v_1000/gui.12368496478808544922.tmp.c:11359: by main__main
/tmp/v_1000/gui.12368496478808544922.tmp.c:11758: by main
```
if change `[]Layoutable` to `[]&Layoutable` then  (2)
```
=================
/tmp/v_1000/gui.5902536158928048214.tmp.c:11391: error: struct or union expected
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.
````

--- a/vlib/v/ast/table.v
+++ b/vlib/v/ast/table.v
@@ -446,6 +446,18 @@ pub fn (t &Table) find_method_with_embeds(sym &TypeSymbol, method_name string) ?
 	}
 }
 
+pub fn (t &Table) get_methods_with_embeds(sym &TypeSymbol) []Fn {
+	mut methods := []Fn{}
+	if sym.info is Struct {
+		for embed in sym.info.embeds {
+			embed_sym := t.sym(embed)
+			methods << embed_sym.methods
+			methods << t.get_methods_with_embeds(embed_sym)
+		}
+	}
+	return methods
+}
+
 fn (t &Table) register_aggregate_field(mut sym TypeSymbol, name string) ?StructField {
 	if sym.kind != .aggregate {
 		t.panic('Unexpected type symbol: $sym.kind')

--- a/vlib/v/ast/table.v
+++ b/vlib/v/ast/table.v
@@ -446,13 +446,13 @@ pub fn (t &Table) find_method_with_embeds(sym &TypeSymbol, method_name string) ?
 	}
 }
 
-pub fn (t &Table) get_methods_with_embeds(sym &TypeSymbol) []Fn {
+pub fn (t &Table) get_embed_methods(sym &TypeSymbol) []Fn {
 	mut methods := []Fn{}
 	if sym.info is Struct {
 		for embed in sym.info.embeds {
 			embed_sym := t.sym(embed)
 			methods << embed_sym.methods
-			methods << t.get_methods_with_embeds(embed_sym)
+			methods << t.get_embed_methods(embed_sym)
 		}
 	}
 	return methods

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -6797,7 +6797,7 @@ static inline __shared__$interface_name ${shared_fn_name}(__shared__$cctype* x) 
 				}
 				else {}
 			}
-			t_methods := g.table.get_methods_with_embeds(st_sym)
+			t_methods := g.table.get_embed_methods(st_sym)
 			for t_method in t_methods {
 				if t_method.name !in method_names {
 					methods << t_method

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -6797,16 +6797,13 @@ static inline __shared__$interface_name ${shared_fn_name}(__shared__$cctype* x) 
 				}
 				else {}
 			}
-			if st_sym.info is ast.Struct {
-				for embed in st_sym.info.embeds {
-					embed_sym := g.table.sym(embed)
-					for embed_method in embed_sym.methods {
-						if embed_method.name !in method_names {
-							methods << embed_method
-						}
-					}
+			t_methods := g.table.get_methods_with_embeds(st_sym)
+			for t_method in t_methods {
+				if t_method.name !in method_names {
+					methods << t_method
 				}
 			}
+
 			for method in methods {
 				mut name := method.name
 				if inter_info.parent_type.has_flag(.generic) {

--- a/vlib/v/tests/interface_struct_with_multi_nested_embed_test.v
+++ b/vlib/v/tests/interface_struct_with_multi_nested_embed_test.v
@@ -1,0 +1,91 @@
+fn test_interface_struct_with_multi_nested_embed() {
+	mut win := &Window{}
+	mut ll := &LinearLayout{}
+	mut lbl := &Label{
+		x: 10
+		y: 20
+	}
+
+	ll.add(mut lbl)
+	win.add(mut ll)
+
+	win.init()
+}
+
+[heap]
+pub struct Window {
+mut:
+	initables []&Initable
+}
+
+interface Initable {
+mut:
+	init(&Window)
+}
+
+pub fn (mut w Window) add(mut initable Initable) {
+	w.initables << initable
+}
+
+interface Container {
+mut:
+	layout()
+}
+
+fn (mut w Window) init() {
+	for wd in w.initables {
+		if mut wd is Container {
+			mut c := wd as Container
+			c.layout()
+		}
+	}
+}
+
+pub struct Rect {
+mut:
+	x int
+	y int
+}
+
+pub fn (r Rect) get_pos() (int, int) {
+	return r.x, r.y
+}
+
+pub struct LayoutBase {
+	Rect
+}
+
+pub struct Base {
+	LayoutBase
+}
+
+pub fn (mut b Base) init(window &Window) {}
+
+[heap]
+pub struct Label {
+	Base
+}
+
+pub interface Layoutable {
+	get_pos() (int, int)
+}
+
+[heap]
+pub struct LinearLayout {
+	Base
+mut:
+	layoutables []Layoutable
+}
+
+pub fn (mut ll LinearLayout) add(mut l Layoutable) {
+	ll.layoutables << l
+}
+
+pub fn (mut ll LinearLayout) layout() {
+	for mut wl in ll.layoutables {
+		x, y := wl.get_pos()
+		println('$x, $y')
+		assert x == 10
+		assert y == 20
+	}
+}

